
From: Davi Arnaut <davi.arnaut@gmail.com>

Data allocated with "__getname()" should always be free'd with "__putname()"
because of the AUDITSYSCALL code.

Signed-off-by: Davi Arnaut <davi.arnaut@gmail.com>
Cc: Urban Widmark <urban@teststation.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 fs/smbfs/symlink.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff -puN fs/smbfs/symlink.c~smbfs-names_cache-memory-leak fs/smbfs/symlink.c
--- devel/fs/smbfs/symlink.c~smbfs-names_cache-memory-leak	2005-11-04 21:44:55.000000000 -0800
+++ devel-akpm/fs/smbfs/symlink.c	2005-11-04 21:44:55.000000000 -0800
@@ -45,7 +45,7 @@ static void *smb_follow_link(struct dent
 		int len = smb_proc_read_link(server_from_dentry(dentry),
 						dentry, link, PATH_MAX - 1);
 		if (len < 0) {
-			putname(link);
+			__putname(link);
 			link = ERR_PTR(len);
 		} else {
 			link[len] = 0;
@@ -59,7 +59,7 @@ static void smb_put_link(struct dentry *
 {
 	char *s = nd_get_link(nd);
 	if (!IS_ERR(s))
-		putname(s);
+		__putname(s);
 }
 
 struct inode_operations smb_link_inode_operations =
_
